### PR TITLE
Wm add ion writer

### PR DESCRIPTION
**Pull request by wmaryszczak**

### CODE REVIEW

File: Anixe.Ion/IIonWriter.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace Anixe.Ion
+{
+    public interface IIonWriter : IDisposable
+    {
+        void WriteSection(string name);
+        void WriteProperty(string name, string value);
+        void WriteProperty(string name, int value);
+        void WriteProperty(string name, double value);
+        void WriteProperty(string name, float value);
```
Comment (lechu445) [inline Anixe.Ion/IIonWriter.cs:13]: class IonWriter also implements `void WriteProperty(string name, bool value);` and `void WriteProperty(string name, char value);`

File: Anixe.Ion/IonWriter.cs [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+            this.state |= WriterState.Property;
+        }
+
+        public void WriteProperty(string name, decimal value)
+        {
+            ValidateWriteProperty(name);
+            ClearState();
+            this.tw.Write(name);
+            this.tw.Write(Consts.IonSpecialChars.EqualsCharacter);
+            this.tw.WriteLine(value.ToString(CultureInfo.InvariantCulture));
```
Comment (lechu445) [inline Anixe.Ion/IonWriter.cs]: The code below uses TextWriter's IFormatProvider:
```
this.tw.Write(value);
this.tw.WriteLine();
```

File: Anixe.Ion/IonWriter.cs [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+            {
+                throw new InvalidOperationException("Only section can be at the top of document");
+            }
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentNullException("Property name must be provided");
+            }
+        }
+
+        public void WriteTableHeader(string[] columns)
```
Comment (lechu445) [inline Anixe.Ion/IonWriter.cs]: just cosmetic suggestion: `public void WriteTableHeader(params string[] columns)`

File: Anixe.Ion/IonWriter.cs [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+            }
+            if (columns == null || columns.Length == 0)
+            {
+                throw new ArgumentNullException("Cannot create empty table header");
+            }
+        }
+
+        public void WriteTableRow(string[] data)
+        {
+            if (this.lastTableColumns == null && !this.state.HasFlag(WriterState.TableHeader))
```
Comment (lechu445) [inline Anixe.Ion/IonWriter.cs]: HasFlag(...) can be replaced with faster implementation regarding to this article http://grahammoore.com/improving-enum-hasflag/

File: Anixe.Ion/IonWriter.cs (hunk 2 above)
Comment (wmaryszczak) [reply Anixe.Ion/IonWriter.cs]: I think we should force writing double with dot. This is part of ion format

File: Anixe.Ion/IIonWriter.cs (hunk 1 above)
Comment (wmaryszczak) [reply Anixe.Ion/IIonWriter.cs:13]: done

File: Anixe.Ion/IonWriter.cs (hunk 4 above)
Comment (wmaryszczak) [reply Anixe.Ion/IonWriter.cs]: you cannot trust anyone :) fixed

File: Anixe.Ion/IonWriter.cs (hunk 3 above)
Comment (wmaryszczak) [reply Anixe.Ion/IonWriter.cs]: done

**Pull request merged**